packages/ontario-design-system-component-library/src/components/ontario-task/ontario-task.tsx

```diff
@@ -0,0 +1,230 @@
+	 */
+	@Listen('setAppLanguage', { target: 'window' })
+	handleSetAppLanguage(event: CustomEvent<Language>) {
+		if (!this.language) {
+			this.language = validateLanguage(event);
+		}
+	}
+
+	/**
+	 * Listen for header language toggling events to update the language dynamically.
+	 */
+	@Listen('headerLanguageToggled', { target: 'window' })
+	handleHeaderLanguageToggled(event: CustomEvent<Language>) {
+		this.language = validateLanguage(event);
+	}
+
+	/**
+	 * Determines the badge colour based on the current `taskStatusState`.
+	 *
+	 * @returns {TaskBadgeColour}
+	 */
+	private getBadgeColour(): TaskBadgeColour {
+		return TaskToBadgeColour[this.taskStatusState] || 'grey';
+	}
+
+	/**
+	 * Provides a translated task status string based on the current language.
+	 */
+	private getTranslatedTaskStatus(): string {
+		const resolvedLanguage = validateLanguage(this.language);
+		return translations.taskStatus[this.taskStatusState][resolvedLanguage];
+	}
+
+	/**
+	 * Renders hint text if available.
+	 */
+	private renderHintText() {
+		if (this.internalHintText) {
+			return (
+				<ontario-hint-text
+					hint={this.internalHintText.hint}
+					hintContentType={this.internalHintText.hintContentType}
+					ref={(el) => (this.hintTextRef = el)}
+				></ontario-hint-text>
+			);
+		}
+		return null;
+	}
+
+	private getClass(): string {
+		return this.hintText ? `ontario-task__label ontario-task__hint-text--true` : `ontario-task__label`;
+	}
+
+	/**
+	 * Renders the task label and status content.
+	 */
+	private renderTaskContent() {
+		return (
+			<div class="ontario-task__content">
+				<h3 id="task-label" class={this.getClass()}>
+					{this.label}
+				</h3>
+				{this.taskStatusState && (
+					<ontario-badge
+						class="ontario-task__badge"
+						role="status"
+						aria-label={`${
+							translations.taskStatus.taskStatus[validateLanguage(this.language)]
+						} ${this.getTranslatedTaskStatus()} (${this.taskStatusState})`}
+						colour={this.getBadgeColour()}
+					>
+						{this.getTranslatedTaskStatus()}
+					</ontario-badge>
+				)}
+			</div>
+		);
+	}
+
+	async componentDidLoad() {
+		this.hintTextId = await this.hintTextRef?.getHintTextId();
+	}
+
+	componentWillLoad() {
+		this.parseHintText();
+		this.language = validateLanguage(this.language);
+		this.validateTaskStatus();
+	}
+
+	render() {
+		const isLinkActive = this.link && !this.deactivateLink;
+		return (
+			<article class="ontario-task" role="group" aria-labelledby="task-label" aria-describedby={this.hintTextId}>
+				{isLinkActive ? (
+					<a
+						href={this.link}
+						target="_blank"
+						rel="noopener noreferrer"
+						class="ontario-task__link"
+						aria-label={`${this.label} ${translations.newWindow[validateLanguage(this.language)]}`}
+					>
+						{this.renderTaskContent()}
+						{this.renderHintText()}
+					</a>
+				) : (
+					<div>
+						{this.renderTaskContent()}
+						{this.renderHintText()}
+					</div>
+				)}
+			</article>
+		);
+	}
+}
```

Comment:
The logic for rendering link and non-link wrappers is duplicating `renderTaskContent` and `renderHintText`. Can we abstract common content into a variable and use conditional wrapping instead?

So like 
```
const taskContent = (
   <>
       {this.renderTaskContent()}
       {this.renderHintText()}
   </>
);
```
```
<article ...>
    {this.link && !this.deactivateLink ? ( <a...> {taskContent} </a>) 
          : ( <div>{taskContent}</div>
     )}
</article>
```